### PR TITLE
Ev 413

### PR DESCRIPTION
**Pull request by trapprakam**

### CODE REVIEW

**Comment by trapprakam** [comment]
I added Paul to this PR because I will like someone with some Java background to have oversite as well. I was only able to add Paul because Alfredo and Albert are now avals for selection as a reviewer.

**Review by NoahLiot — APPROVED** [review]

File: contribs/src/main/java/com/netflix/conductor/contribs/kafka/resource/handlers/ResourcesLoader.java [hunk 1]
```diff
@@ -95,12 +102,99 @@ private void read(final InputStream inputStream, final String parent) throws IOE
...
             }
         }
     }
+
+    /**
+     * Load an instance of the class specified by the given class name
+     * @param className The name of the class to be loaded
+     * @return An instance of the requested class to be loaded
+     */
+    private <T> Class<T> loadResourceClass(String className) {
```
Comment (weispa) [inline contribs/src/main/java/com/netflix/conductor/contribs/kafka/resource/handlers/ResourcesLoader.java]: make parameters final

File: contribs/src/main/java/com/netflix/conductor/contribs/kafka/resource/handlers/ResourcesLoader.java [hunk 2]
```diff
@@ -95,12 +102,99 @@ private void read(final InputStream inputStream, final String parent) throws IOE
...
+        /**
+         * Provides the functionality to visit the header of the class
+         * @param version The class version
+         * @param access The class access flags
+         * @param name The name of the given class
+         * @param signature The signature of the class if its not a generic class or implements or extends generics
+         * @param superName The name of the given class associated super class, if exist
+         * @param interfaces The name of the given class associated interface classes, if exist
+         */
+        public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
```
Comment (weispa) [inline contribs/src/main/java/com/netflix/conductor/contribs/kafka/resource/handlers/ResourcesLoader.java:166]: same as above, params should be final where possible

File: contribs/src/main/java/com/netflix/conductor/contribs/kafka/resource/handlers/ResourcesLoader.java [hunk 3]
```diff
@@ -95,12 +102,99 @@ private void read(final InputStream inputStream, final String parent) throws IOE
...
+    public Set<Class<?>> getClasses() {
+        return classes;
+    }
+
+    /**
+     * The ClassVisitor class have all the methods to access the components of a given
+     * Java class. This subclass provides extended functions for verifying if a class is
+     * annotated and if so, load the class.
+     */
+    private final class AnnotatedClassesVisitor extends ClassVisitor {
```
Comment (weispa) [inline contribs/src/main/java/com/netflix/conductor/contribs/kafka/resource/handlers/ResourcesLoader.java:141]: inner classes must be static

**Pull request merged**